lib/src/Uncertainty/Model/DistributionImplementation.cxx

```diff
@@ -148,11 +148,23 @@ Distribution DistributionImplementation::operator + (const Distribution & other)
 
 Distribution DistributionImplementation::operator + (const DistributionImplementation & other) const
 {
-  if ((dimension_ != 1) || (other.dimension_ != 1)) throw NotYetImplementedException(HERE) << "In DistributionImplementation::operator + (const DistributionImplementation & other) const: can add 1D distributions only.";
-  Collection< Distribution > coll(2);
-  coll[0] = *this;
-  coll[1] = other.clone();
-  return new RandomMixture(coll);
+  if (dimension_ != other.dimension_) throw InvalidDimensionException(HERE) << "Can only sum distributions with the same dimension:" << dimension_ <<  " vs " << other.dimension_;
+
+  if ((dimension_ > 1) && (!hasIndependentCopula() || !other.hasIndependentCopula())) throw NotYetImplementedException(HERE) << "Can only sum distributions with independent copulas";
+
+  Collection< Distribution > marginals(dimension_);
+  for (UnsignedInteger j = 0; j < dimension_; ++ j)
+  {
+    Collection< Distribution > coll(2);
+    coll[0] = getMarginal(j);
+    coll[1] = other.getMarginal(j);
+    marginals[j] = new RandomMixture(coll);
+  }
+
+  if (dimension_ > 1)
```

Comment:
If you separate the case dimension_==1 from dimension_>1, then you should test it at line 152 and keep the old code for this (very usual) case, with no call to getMarginal().